controllers/platformnetwork_controller.go

```diff
@@ -661,38 +742,291 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
-			instance.Status.ConfigurationUpdated = false
-			instance.Status.StrategyRequired = cloudManager.StrategyNotRequired
-			if instance.Status.DeploymentScope == cloudManager.ScopePrincipal {
-				r.CloudManager.SetResourceInfo(cloudManager.ResourcePlatformnetwork, "", instance.Name, instance.Status.Reconciled, cloudManager.StrategyNotRequired)
+	inSync, err2 := r.UpdateInsyncStatus(client, instance, err)
+	if err2 != nil {
+		return inSync, err2
+	}
+
+	return inSync, err
+}
+
+// Returns poniter to instance of starlingxv1.Host which is used for creating
+// lock / unlock strategies & creating annotation for interface network reconciliation
+// against the host.
+func (r *PlatformNetworkReconciler) GetHostInstance(hostname string, request_namespace string) (*starlingxv1.Host, error) {
+	host_instance := &starlingxv1.Host{}
+	host_namespace := types.NamespacedName{Namespace: request_namespace, Name: hostname}
+	err := r.Client.Get(context.TODO(), host_namespace, host_instance)
+	if err != nil {
+		logPlatformNetwork.Error(err, "Failed to get resource for host namespace %v", host_namespace)
+	}
+	return host_instance, err
+}
+
+// This method handles strategy updates such as lock_required / unlock_required / not_required
+// for the given host.
+func (r *PlatformNetworkReconciler) HandleHostStrategyUpdates(host_strategy cloudManager.HostStrategyInfo, request_namespace string) (err error) {
+	host_instance, err := r.GetHostInstance(host_strategy.Host.Hostname, request_namespace)
+	if err != nil {
+		logPlatformNetwork.Error(err, "Failed to get resource for host namespace %v", request_namespace)
+		return err
+	}
+
+	if host_instance.Status.StrategyRequired != host_strategy.StrategyRequired {
+		host_instance.Status.StrategyRequired = host_strategy.StrategyRequired
+		err := r.Client.Status().Update(context.TODO(), host_instance)
+		if err != nil {
+			logPlatformNetwork.Error(err, "Failed to update status %v", host_instance.Status)
+			return err
+		}
+	}
+
+	if !r.CloudManager.GetStrageySent() {
+		msg := fmt.Sprintf("Applying %s strategy on host %s", host_strategy.StrategyRequired, host_instance.Name)
+		logPlatformNetwork.Info(msg)
+		reconciled := host_instance.Status.Reconciled
+		strategy := host_instance.Status.StrategyRequired
+		r.CloudManager.SetResourceInfo(cloudManager.ResourceHost, host_strategy.Host.Personality, host_instance.Name, reconciled, strategy)
+		m := hostController.NewLockedDisabledHostMonitor(host_instance, host_strategy.Host.ID)
+		_ = r.CloudManager.StartMonitor(m, msg)
+	}
+	return common.NewResourceConfigurationDependency("waiting for system to be in locked state before reconfiguring network")
+}
+
+// ReconcileResource interacts with the system API in order to reconcile the
+// state of a data network with the state stored in the k8s database.
+func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork, request_namespace string) (err error) {
```

Comment:
Expect to break this long method by scenarios: ReconcileResourceBootstrap, ReconcileResourcePincipal(ReconcileResourcePincipalSimplex, ReconcileResourcePincipalMultinodes)

Reply:
> Expect to break this long method by scenarios: ReconcileResourceBootstrap, ReconcileResourcePincipal(ReconcileResourcePincipalSimplex, ReconcileResourcePincipalMultinodes)

Done.